lib/fs.js

```diff
@@ -2070,11 +2071,82 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     } else {
       offset += written;
       length -= written;
-      writeAll(fd, isUserFd, buffer, offset, length, signal, callback);
+      writeAll(
+        fd, isUserFd, buffer, offset, length, signal, encoding, callback);
     }
   });
 }
 
+async function writeAllCustomIterable(
```

Comment:
Do not use `async` with a callback. Mixing those two can only lead to bugs. I would recommend having a promisified version of `fs.write()` and just use async/await.

Reply:
@mcollina Thanks for your suggestion 👍 . I'm curious about the advantages of having a promisified version of `fs.write()` and using `async/await`. Could you elaborate on why this approach might be better than mixing async with a callback? I'm interested in understanding the potential benefits and how it could prevent bugs.